MovingBlocks/TerasologyLauncher
# build(pmd): update PMD rules and fix issues

### Contains

Updated and customized rules for PMD, adjusted for the launcher project.

All issues have been adressed such that we no longer ignore PMD errors and instead will fail a build if new issues occur.

- build(pmd): use custom PDM rules for the launcher
- build(pmd): adjust custom rules

### How to test

Run the PMD checks locally via 

```
gradlew pmdMain
gradlew pmdTest
```

Run the CheckStyle checks locally via 

```
gradlew checkstyleMain
gradlew checkstyleTest
```

Or run everything at once via
```
gradlew check -x test
```

### Outstanding before merging

- [x] fix reported PMD issues or suppress warnings
- [x] fix reported CheckStyle issues or suppress warnings
